### ISSUE TITLE
everything overwritten on mupx deploy

### ISSUE THREAD

**Issue by crackedbeefcake** [arunoda/meteor-up-legacy#967]
I am using the `ostrio:files` package to upload images. everything works fine, but after i update the app the mupx deploy, all my images are gone. 
I store them at 
`Images = new Meteor.Files({
    collectionName: 'Images',
    storagePath: '/data/UploadedFiles',
    debug: true
});`

**Comment by crackedbeefcake** [comment]
[I already had a lenghty and detailed conversation about this with the files package author ](https://github.com/VeliovGroup/Meteor-Files/issues/72)

**Comment by ipa1981** [comment]
I also found my images lost. Didn't realize it was because of mupx, but now it seems we're on the same boat. so +1

**Comment by crackedbeefcake** [comment]
i assume its mupx. on localhost the files persist a reboot.

**Comment by MasterJames** [comment]
The port over at kadirahq I think makes it easier to add another "volume" to the docker run command that similarly mounts a folder (outside the docker contsiner) for mongdb's data that can persist redeployment.

**Comment by crackedbeefcake** [comment]
The data isnt safed inside mongo. its safed to the filesystem. thats `ostrio:files`. and the mongodb persists the `mupx deploy`. just the files are vaped.

**Comment by MasterJames** [comment]
That's way you have to ask the docker container at runtime to mount a filesystem point with the volume command.
Here is where it's done for mongodb. ..https://github.com/arunoda/meteor-up/blob/mupx/templates/linux/start.sh#L26
I believe the new version provides a setting to add other "volumes".
Maybe that's to be done via a custom container image?  Better to ask there I guess.

**Comment by crackedbeefcake** [comment]
To get this straight: mupx internally uses docker to run meteor on the server. on my machine, my app gets bundled, sent to the server and run by docker with the help of the preexisting meteor image `meteorhacks/meteord:base`.

and i cant access the Images I have safed after  `mupx deploy` because [docker rm -f $APPNAME](https://github.com/arunoda/meteor-up/blob/mupx/templates/linux/start.sh#L11) gets run, which destroys the folder where the Images where safed.

or is it, bacause the new docker doesnt have access to the filesystem point where the Images are safed?

**Comment by MasterJames** [comment]
Yes sounds right it gets destroyed with the container unless you explicitly ask it to mount a safe folder from the target via a docker run "volume" attribute. I suppose this should be an S3 source or something safe from the target VMs creation scheme.

**Comment by ipa1981** [comment]
Master,  sorry,  but shouldn't the tool be designed to make life easier? I believe there are majority of developers who has file uploads on their servers and looking for way of easy deploy,  with as less server configuration as possible. 
The mup,  which was used before I upgraded to mupx,  hadn't this problem and worked out of the box.

**Comment by crackedbeefcake** [comment]
Its true, it would be really cool to not have this problem. 
@MasterJames you say this has to be a S3 source. 
Can the safe volume also be on the same server as the app? and if, how?

@ipa1981 what mup are you using? ...where using, when you didn't have this problem?

**Comment by ipa1981** [comment]
@FinnFrotscher I'm using mupx.

**Comment by ipa1981** [comment]
Otherwise, if it is really hard to solve this issue - maybe someone can update documentation:
- Describe what should be done to keep local storage safe with mupx
- Put a notice in docs, that mupx is not compatible to upgrade from mup and files needs to be kept safe way somehow. In my case I lost just 10 images. I believe in worse cases someone could lose much more.

**Comment by crackedbeefcake** [comment]
but you did use mup before and it worked with Meteor-Files? @ipa1981 
why did you switch?

**Comment by ipa1981** [comment]
@FinnFrotscher :
I've used these packages, maybe not the same, but also stores files in server folder:
tomi:upload-jquery           2.4.0  Client template for uploads using "jquery-file-upload" from blueimp
tomi:upload-server           1.3.4  Upload server for Meteor. Allows to save and serve files from arbitrary directory

Yes, it worked well with old mup until I've upgraded to mupx.

**Comment by MasterJames** [comment]
Always test after R&D with your GuineaPig machines. measure twice cut once. Note: the new version at kadirahq/meteor-up is the way forward this version is depreciated. Everybody's needs are different.
If you have space you can even run full meteor without deployment. I'm waiting for EFS on AWS to leave Preview and use mupx as a learning tool, while designing a new tool for my unique needs.
What you'll say sounds workable. Maybe report back your preferences and final layout? Goodluck.

**Comment by MasterJames** [comment]
Maybe there's a completely different approach that is useful for some projects?  Google API etc.
https://developers.google.com/drive/v3/web/quickstart/nodejs

**Comment by Nauzer** [comment]
My suggestion is indeed to connect to - for example - and S3 instance for your uploaded files.
However if you don't want to you can add two lines to the bottom of your start.sh

[NODE_ROOT]/node_modules/mupx/template/linux/start.sh

UPLOAD_FOLDER = '/your/upload/folder'
cp -r $APP_PATH/last/bundle/$UPLOAD_FOLDER $BUNDLE_PATH/bundle/

**Comment by dr-dimitru** [comment]
Hello everyone, as I see you already figured out how to use Meteor-Files with MUPX, safely to avoid data loss.

Could any of you post (_or send a PR_) little note about MUPX usage within Meteor-Files, so I'll add it to Readme.md and other docs?

**Comment by MasterJames** [comment]
This version is depreciated only the new version at kadirahq meteor-up is accepting PRs.

**Comment by dr-dimitru** [comment]
@MasterJames I meant PR to [_MF_](https://github.com/VeliovGroup/Meteor-Files)
